src/w3d/lib/gcd_lcm.cpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Greatest Common Divisor/Lowest Common Multiple
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "gcd_lcm.h"
+
+unsigned int Greatest_Common_Divisor(unsigned int a, unsigned int b)
```

Comment:
What are these being used for? I suggest moving them into the header and making them `constexpr` as they can be computed at compile time for any fixed values of a and b.

Reply:
Its used by the particle emitter code and the values passed in are not constants. I took another look at this in a few binaries and discovered evidence about how it actually looked so I changed it.

Reply:
The way these 2 functions are now is how they look in zero hour worldbuilder so they should be good.